src/NexusMods.CrossPlatform/Process/ProcessFactory.cs

```diff
@@ -25,4 +25,28 @@ public ProcessFactory(ILogger<ProcessFactory> logger)
         _logger.LogInformation("Executing command `{Command}`", command.ToString());
         return await command.ExecuteAsync(cancellationToken);
     }
+
+    /// <inheritdoc />
+    public Task ExecuteProcessAsync(System.Diagnostics.Process process, CancellationToken cancellationToken = default)
+    {
+        var tcs = new TaskCompletionSource(TaskCreationOptions.LongRunning);
+
+        process.Exited += (sender, args) =>
+        {
+            tcs.SetResult();
+            process.Dispose();
+        };
+        
+        cancellationToken.Register(() =>
+        {
+            if (!process.HasExited)
+            {
+                process.Kill();
+            }
+        });
+
+        process.Start();
+
+        return tcs.Task; 
+    }
```

Comment:
Missing `process.EnableRaisingEvents = true`. Without this, the `Exited` event won't ever been raised. This method also needs some `try/catch` and some logging.